### PR TITLE
lcov: add recipe

### PR DESCRIPTION
**Pull request by Max-Might**
I have created a recipe for lcov.

### CODE REVIEW

File: dev-util/lcov/lcov-1.12.recipe [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+function and branch coverage measurement."
+HOMEPAGE="http://ltp.sourceforge.net/coverage/lcov.php"
+SOURCE_URI="https://github.com/linux-test-project/lcov/releases/download/v1.12/lcov-1.12.tar.gz"
+CHECKSUM_SHA256="b474e49c6c962754063b3be97a757a2ba9e7a455f0aea612863bf67e9b8b8ea7"
+COPYRIGHT="2002 - 2016 The lcov Developers"
+LICENSE="GNU GPL v2"
+REVISION="1"
+PATCHES="lcov-1.12.patchset"
+
+ARCHITECTURES="x86"
```
Comment (waddlesplash) [inline dev-util/lcov/lcov-1.12.recipe]: If there are no arch-specific files, this can be "any".

Comment (Max-Might) [inline dev-util/lcov/lcov-1.12.recipe]: It does not work with `gcov` provided by gcc2.

Comment (waddlesplash) [inline dev-util/lcov/lcov-1.12.recipe]: Yes; but that's not really this's fault, that's GCC2's fault. So I say make it "any".

**Comment by Max-Might** [comment]
Done.

File: dev-util/lcov/lcov-1.12.recipe (hunk 1 above)
Comment (Max-Might) [inline dev-util/lcov/lcov-1.12.recipe]: I have updated the 'architectures' field to 'any'.

**Pull request merged**